src/main/java/org/folio/rs/service/AccessionQueueService.java

```diff
@@ -82,6 +108,69 @@ public void processAccessionQueueRecord(List<DomainEvent> events) {
     });
   }
 
+  public AccessionQueue processPostAccession(AccessionRequest accessionRequest) {
+    var locationMapping = getLocationMapping(accessionRequest);
+    var item = getItem(accessionRequest);
+    var holding = holdingsStorageClient.getHoldingsRecordsByQuery("id==" + item.getHoldingsRecordId())
+      .getResult().get(0);
+    var instance = inventoryClient.getInventoryInstancesByQuery("id==" + holding.getInstanceId())
+      .getResult().get(0);
+
+    var accessionQueueRecord = buildAccessionQueueRecord(item, instance, locationMapping);
+    accessionQueueRepository.save(accessionQueueRecord);
+
+    item.setPermanentLocation(new InventoryItemPermanentLocation().id(locationMapping.getFolioLocationId()));
+    inventoryClient.putInventoryItem(item.getId(), item);
+
+    if (!item.getPermanentLocation().getId().equals(holding.getPermanentLocationId())) {
```

Comment:
Could you please simplify/refactor this method for better readability? My personal recommendations (if applicable): extract logic from if-blocks to separate methods, extract if statements to separate methods.